Application/src/Parser.java

```diff
@@ -0,0 +1,135 @@
+
+        try {
+            diagramObject parsedDiagram;
+            parsedDiagram = gson.fromJson(inputJSON, diagramObject.class);
+
+            // get the Meta data
+            diagramObject.Meta metaElement = parsedDiagram.getMeta();
+
+            // get the diagram type
+            parsedDiagram.getType();
+
+            // get the processes which contains the class names
+            for (diagramObject.Processes processesElement : parsedDiagram.getProcesses()) {
+                processes += "'" + processesElement.getSequenceDiagramClass() + ":" + processesElement.getName() + "',";
+                properProcesses = (processes.substring(0, processes.length() - 1));
+                // yield only the names of the classes
+                classNames += processesElement.getName() + ",";
+                properClassNames = (classNames.substring(0, classNames.length() - 1));
+            }
+
+            // get the elements of the First Diagram
+            for (diagramObject.ContentArray diagramElement : parsedDiagram.getDiagram().getContent().get(0).getContent()) {
+                firstDiagram += "{" + diagramElement.getFrom() + "," + diagramElement.getTo() + ",[" + diagramElement.getMessage().get(0) + ", " + diagramElement.getMessage().get(1) + ", " + diagramElement.getMessage().get(2) + "]" + "}" + ",";
+                properFirstDiagram = (firstDiagram.substring(0, firstDiagram.length() - 1));
+            }
+
+
+            // get the elements of the Parallel Diagram
+            for (diagramObject.ContentArray parallelDiagramElement : parsedDiagram.getDiagram().getContent().get(1).getContent()) {
+                parallelDiagram += "{" + parallelDiagramElement.getFrom() + "," + parallelDiagramElement.getTo() + ",[" + parallelDiagramElement.getMessage().get(0) + "]" + "}" + ",";
+                properParallelDiagram = (parallelDiagram.substring(0, parallelDiagram.length() - 1));
+            }
+        }
+
+        catch (NullPointerException e) {
+            syntaxErrorMessage();
+
+        }
+    }
+
+    // displays a frame notifying the user that the imported file contains unrecognisable syntax
+    private static void syntaxErrorMessage() {
+        Alert alert = new Alert(Alert.AlertType.INFORMATION);
+        alert.setTitle("Import failed");
+        alert.setHeaderText("Unknown syntax");
+        alert.setContentText("The selected file contains unrecognisable syntax");
+        alert.show();
+    }
+
+    /**
+     * gives a String containing the first diagram to be handled by the backend
+     * @return FirstSequenceDiagram which contains a counter, the processes, the class names, the first diagram's messages and content.
+     */
+
+    public static String getFirstSequenceDiagram(){
+
+        String FirstSequenceDiagram = "{" + counter + ",[" + properProcesses + "]," + "["+ properClassNames +"],"+ "["+ properFirstDiagram + "]}";
+        // increase the counter by 2 since it is always an even number
```

Comment:
Why is what always an even number? (OBS: this is a question because I don't understand and got curious, not necessarily anything wrong with the code :D )

Reply:
@IsabelleTornqvist  defining two different counter variables (one is even and the other is odd) that will increment each time the diagram calling method that they are is called in order to differentiate between the imported diagrams